test/e2e/pivoting_test.go

```diff
@@ -24,146 +26,210 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const bmoPath = "BMOPATH"
+const (
+	bmoPath                    = "BMOPATH"
+	ironicTLSSetup             = "IRONIC_TLS_SETUP"
+	ironicBasicAuth            = "IRONIC_BASIC_AUTH"
+	Kind                       = "kind"
+	NamePrefix                 = "NAMEPREFIX"
+	restartContainerCertUpdate = "RESTART_CONTAINER_CERTIFICATE_UPDATED"
+	ironicNamespace            = "IRONIC_NAMESPACE"
+)
+
+type PivotingInput struct {
+	E2EConfig             *clusterctl.E2EConfig
+	BootstrapClusterProxy framework.ClusterProxy
+	TargetCluster         framework.ClusterProxy
+	SpecName              string
+	ClusterName           string
+	Namespace             string
+	ArtifactFolder        string
+	ClusterctlConfigPath  string
+}
 
-func pivoting() {
+func pivoting(ctx context.Context, inputGetter func() PivotingInput) {
 	Logf("Starting pivoting tests")
-	listBareMetalHosts(ctx, bootstrapClusterProxy.GetClient(), client.InNamespace(namespace))
-	listMetal3Machines(ctx, bootstrapClusterProxy.GetClient(), client.InNamespace(namespace))
-	listMachines(ctx, bootstrapClusterProxy.GetClient(), client.InNamespace(namespace))
-	listNodes(ctx, targetCluster.GetClient())
+	input := inputGetter()
+	numberOfWorkers := int(*input.E2EConfig.GetInt32PtrVariable("WORKER_MACHINE_COUNT"))
+	numberOfControlplane := int(*input.E2EConfig.GetInt32PtrVariable("CONTROL_PLANE_MACHINE_COUNT"))
+	numberOfAllBmh := numberOfWorkers + numberOfControlplane
+
+	listBareMetalHosts(ctx, input.BootstrapClusterProxy.GetClient(), client.InNamespace(input.Namespace))
+	listMetal3Machines(ctx, input.BootstrapClusterProxy.GetClient(), client.InNamespace(input.Namespace))
+	listMachines(ctx, input.BootstrapClusterProxy.GetClient(), client.InNamespace(input.Namespace))
+	listNodes(ctx, input.TargetCluster.GetClient())
 
 	By("Remove Ironic containers from the source cluster")
 	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
-	if ephemeralCluster == KIND {
-		removeIronicContainers()
-	} else {
-		removeIronicDeployment()
+	isDeployment := true
```

Comment:
can we rename this to be more specific, i.e: `isIronicDeployment` or similar?

Reply:
changed to `isIronicDeployment`